hack/istio-install-openshift.sh

```diff
@@ -7,41 +7,62 @@ if [ "$?" != 0 ]; then
   exit 1
 fi
 
+if [ "$1" = "" ]; then
+  echo "Will download and install the latest Istio."
+  echo "If you want a specific version, pass the version as an argument to this script."
+else
+  echo "Will download and install Istio version $1"
+fi
+
 # put istio files in the output directory
 INSTALL_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )/_output/istio"
 mkdir -p $INSTALL_DIR
 cd $INSTALL_DIR
 
 # download istio
-curl -L https://git.io/getLatestIstio | sh -
+curl -L https://git.io/getLatestIstio | ISTIO_VERSION=$1 sh -
 cd ${INSTALL_DIR}/istio-*
 ISTIO_DIR="$(pwd)"
 echo "Istio has been downloaded here: $ISTIO_DIR"
 echo "You may want to put its bin/ directory in your path to pick up istioctl"
 
-
 # from https://github.com/redhat-developer-demos/istio-tutorial
-oc adm policy add-scc-to-user anyuid -z istio-ingress-service-account -n istio-system
-oc adm policy add-scc-to-user anyuid -z istio-egress-service-account -n istio-system
-oc adm policy add-scc-to-user anyuid -z default -n istio-system
-oc create -f install/kubernetes/istio.yaml
+# from https://blog.openshift.com/evaluate-istio-openshift/
+# from https://istio.io/docs/setup/kubernetes/quick-start.html
+
+oc new-project istio-system
 oc project istio-system
+oc adm policy add-scc-to-user anyuid -z istio-ingress-service-account
+oc adm policy add-scc-to-user privileged -z istio-ingress-service-account
+oc adm policy add-scc-to-user anyuid -z istio-egress-service-account
+oc adm policy add-scc-to-user privileged -z istio-egress-service-account
+oc adm policy add-scc-to-user anyuid -z istio-pilot-service-account
+oc adm policy add-scc-to-user privileged -z istio-pilot-service-account
+oc adm policy add-scc-to-user anyuid -z default
+oc adm policy add-scc-to-user privileged -z default
+oc adm policy add-cluster-role-to-user cluster-admin -z default
+oc adm policy add-scc-to-user anyuid -z istio-grafana-service-account
+oc adm policy add-scc-to-user privileged -z istio-pilot-service-account
+oc adm policy add-scc-to-user anyuid -z istio-prometheus-service-account
+oc adm policy add-scc-to-user privileged -z istio-prometheus-service-account
+
+oc apply -f install/kubernetes/istio.yaml
 oc expose svc istio-ingress
-oc apply -f install/kubernetes/addons/prometheus.yaml
-oc apply -f install/kubernetes/addons/grafana.yaml
-oc apply -f install/kubernetes/addons/servicegraph.yaml
+
+oc create -f install/kubernetes/addons/prometheus.yaml
+oc create -f install/kubernetes/addons/grafana.yaml
+oc create -f install/kubernetes/addons/servicegraph.yaml
```

Comment:
No blocker tip:
I normally skip the servicegraph as it is not really necessary and in some versions doesn't deploy correctly leaving the pod in crash state.